File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsShopMapper.java
package top.charjin.shoppingserver.mapper;

import top.charjin.shoppingserver.entity.OsShop;

import java.util.List;

public interface OsShopMapper {
    int deleteByPrimaryKey(Integer id);

    int insert(OsShop record);

    int insertSelective(OsShop record);

    OsShop selectByPrimaryKey(Integer id);

    int updateByPrimaryKeySelective(OsShop record);

    int updateByPrimaryKey(OsShop record);

    List<OsShop> selectShopByUserId(int userId);
}

File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsShopMapper.kt
package top.charjin.shoppingserver.mapper

import top.charjin.shoppingserver.entity.OsShop

interface OsShopMapper {
    fun deleteByPrimaryKey(id: Int?): Int

    fun insert(record: OsShop): Int

    fun insertSelective(record: OsShop): Int

    fun selectByPrimaryKey(id: Int?): OsShop

    fun updateByPrimaryKeySelective(record: OsShop): Int

    fun updateByPrimaryKey(record: OsShop): Int

    fun selectShopByUserId(userId: Int): List<OsShop>
}